src/main/java/com/aws/greengrass/device/session/SessionConfig.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.device.session;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.util.Coerce;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SessionConfig {
+    private static final Logger LOGGER = LogManager.getLogger(SessionConfig.class);
+    // arbitrarily set default
+    protected static final int DEFAULT_SESSION_CAPACITY = 100_000;
+    public static final String CLIENT_DEVICE_AUTH_SESSION_CAPACITY_TOPIC = "clientDeviceAuthSessionCapacity";
+
+    private final AtomicInteger clientDeviceAuthSessionCapacity = new AtomicInteger(DEFAULT_SESSION_CAPACITY);
+
+    private final Topics configuration;
+
+    /**
+     * Constructor.
+     *
+     * @param configuration Configuration topic for this service
+     */
+    public SessionConfig(Topics configuration) {
+        this.configuration = configuration;
+        this.clientDeviceAuthSessionCapacity.set(getConfiguredSessionCapacity());
+
+        this.configuration.subscribe((whatHappened, node) -> {
+            // update session capacity to the latest configured value
+            updateSessionCapacity(getConfiguredSessionCapacity());
+            LOGGER.atInfo().kv("clientDeviceAuthSessionCapacity", clientDeviceAuthSessionCapacity.get())
+                    .log("Configuration updated");
+        });
+    }
+
+    /**
+     * Get configured Client-Device-Auth Session capacity.
+     *
+     * @return session capacity
+     */
+    public int getSessionCapacity() {
+        return clientDeviceAuthSessionCapacity.get();
+    }
+
+    /**
+     * Updates Client-Device-Auth Session capacity to the desired int value.
+     *
+     * @param newCapacity desired Client-Device-Auth Session capacity
+     */
+    public void updateSessionCapacity(int newCapacity) {
+        this.clientDeviceAuthSessionCapacity.set(newCapacity);
```

Comment:
```suggestion
        clientDeviceAuthSessionCapacity.set(newCapacity);
```
nit: for consistency w/ `getSessionCapacity`